### PR TITLE
Update In-Source Documentation

### PR DESCRIPTION
**Pull request by ElektrojungeAtWork**
This PR includes the following updates:
- Fix a few typos.
- ~Don't use title case for "Swift package manager".~
- Use Apple-style docs for abstracts, e.g. use 3rd person for verb phrases ("Creates..." instead of "Create").
- Add source docs for package resources, localized resources, and binary packages.

### CODE REVIEW

File: Sources/PackageDescription/BuildSettings.swift [hunk 1]
```diff
@@ -131,9 +131,9 @@ public struct CSetting: Encodable {
         return CSetting(name: "define", value: [settingValue], condition: condition)
     }
 
-    /// Set unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
+    /// Sets unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
     ///
-    /// As the usage of the word "unsafe" implies, the Swift Package Manager
```
Comment (neonichu) [inline Sources/PackageDescription/BuildSettings.swift:136]: Is this right? Looking at swift.org, it seems like we are using the spelling "Swift Package Manager"

File: Sources/PackageDescription/BuildSettings.swift [hunk 2]
```diff
@@ -195,9 +195,9 @@ public struct CXXSetting: Encodable {
         return CXXSetting(name: "define", value: [settingValue], condition: condition)
     }
 
-    /// Set unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
+    /// Sets unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
     ///
-    /// As the usage of the word "unsafe" implies, the Swift Package Manager
```
Comment (neonichu) [inline Sources/PackageDescription/BuildSettings.swift:200]: same as above

File: Sources/PackageDescription/BuildSettings.swift [hunk 3]
```diff
@@ -245,9 +245,9 @@ public struct SwiftSetting: Encodable {
         return SwiftSetting(name: "define", value: [name], condition: condition)
     }
 
-    /// Set unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
+    /// Sets unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
     ///
-    /// As the usage of the word "unsafe" implies, the Swift Package Manager
```
Comment (neonichu) [inline Sources/PackageDescription/BuildSettings.swift:250]: same as above

File: Sources/PackageDescription/BuildSettings.swift [hunk 4]
```diff
@@ -303,9 +303,9 @@ public struct LinkerSetting: Encodable {
         return LinkerSetting(name: "linkedFramework", value: [framework], condition: condition)
     }
 
-    /// Set unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
+    /// Sets unsafe flags to pass arbitrary command-line flags to the corresponding build tool.
     ///
-    /// As the usage of the word "unsafe" implies, the Swift Package Manager
```
Comment (neonichu) [inline Sources/PackageDescription/BuildSettings.swift:308]: same as above

File: Sources/PackageDescription/PackageDependency.swift [hunk 5]
```diff
@@ -207,9 +207,9 @@ extension Package.Dependency {
     }
 
   #if !PACKAGE_DESCRIPTION_4
-    /// Add a dependency to a local package on the filesystem.
+    /// Adds a package dependency to a local package on the filesystem.
     ///
-    /// The Swift Package Manager uses the package dependency as-is
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDependency.swift:212]: same as above

File: Sources/PackageDescription/PackageDependency.swift [hunk 6]
```diff
@@ -222,14 +222,16 @@ extension Package.Dependency {
         return .init(name: nil, url: path, requirement: ._localPackageItem)
     }
 
-    /// Add a dependency to a local package on the filesystem.
+    /// Adds a package dependency to a local package on the filesystem.
     ///
-    /// The Swift Package Manager uses the package dependency as-is
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDependency.swift:227]: same as above

File: Sources/PackageDescription/PackageDependency.swift [hunk 7]
```diff
@@ -222,14 +222,16 @@ extension Package.Dependency {
...
-    /// and does not perform any source control access. Local package dependencies
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDependency.swift:228]: just curious: why prefer one over the other? 🤔

File: Sources/PackageDescription/PackageDescription.swift [hunk 8]
```diff
@@ -23,11 +23,12 @@ import Foundation
 /// statement to provide the name of the package, its targets, products,
 /// dependencies, and other configuration options.
 ///
-/// By convention, the properties of a `Package` are defined in a single nested
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDescription.swift:26]: If we drop the code voice, should "package" be lower case?

File: Sources/PackageDescription/PackageDescription.swift [hunk 9]
```diff
@@ -47,32 +48,29 @@ import Foundation
...
+/// followed by a version number such as `// swift-tools-version:5.3.
 ///
-///     // swift-tools-version:3.0.2
-///     // swift-tools-version:3.1
-///     // swift-tools-version:4.0
-///     // swift-tools-version:5.0
-///     // swift-tools-version:5.1
+/// The Swift tools version declares:
 ///
-/// The Swift tools version declares the version of the `PackageDescription`
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDescription.swift:59]: Why are we dropping code voice for `PackageDescription`? It is a module name. I think we should spell it differently if we don't want to reference the module name here.

File: Sources/PackageDescription/PackageDescription.swift [hunk 10]
```diff
@@ -47,32 +48,29 @@ import Foundation
...
+/// without having to update your package manifest and without losing access to
+/// existing packages.
 public final class Package {
 
       /// A package dependency of a Swift package.
       ///
       /// A package dependency consists of a Git URL to the source of the package,
       /// and a requirement for the version of the package.
       ///
-      /// The Swift Package Manager performs a process called *dependency resolution* to
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDescription.swift:75]: same as above

File: Sources/PackageDescription/PackageDescription.swift [hunk 11]
```diff
@@ -185,19 +184,19 @@ public final class Package {
     }
     private var _defaultLocalization: LanguageTag?
 
-    /// The name to use for C Modules.
+    /// The name to use for C modules.
     ///
-    /// If present, the Swift Package Manager searches for a `<name>.pc` file
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDescription.swift:190]: same as above

File: Sources/PackageDescription/PackageDescription.swift [hunk 12]
```diff
@@ -185,19 +184,19 @@ public final class Package {
...
 
-    /// An array of providers for the system target.
+    /// An array of providers for a system target.
     public var providers: [SystemPackageProvider]?
 
-    /// The list of targets.
+    /// The list of targets that are part of this package.
     public var targets: [Target]
 
-    /// The list of products that this package vends and that can be run or used by its clients.
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDescription.swift:200]: I don't think this is correct, clients can't actually run executable products today.

File: Sources/PackageDescription/PackageDescription.swift [hunk 13]
```diff
@@ -218,13 +217,15 @@ public final class Package {
...
     ///
     /// - Parameters:
-    ///     - name: The name of the Swift package.
-    ///     - pkgConfig: Additional flags for a system package.
+    ///     - name: The name of the Swift package or `nil` to deduce the name from the package’s Git URL.
+    ///     - pkgConfig: The name to use for C modules. If present, the Swift 
+    ///           Package Manager searches for a <name>.pc file to get the
+    ///           required additional flags for a system target.
     ///     - providers: The package providers for a system package.
-    ///     - products: The list of products that this package vends and that can be run or used by its clients.
```
Comment (neonichu) [inline Sources/PackageDescription/PackageDescription.swift:227]: same as above

File: Sources/PackageDescription/Product.swift [hunk 14]
```diff
@@ -108,7 +108,7 @@ public class Product: Encodable {
 
         /// The type of the library.
         ///
-        /// If the type is unspecified, the Swift Package Manager automatically
```
Comment (neonichu) [inline Sources/PackageDescription/Product.swift:111]: same as above

File: Sources/PackageDescription/Product.swift [hunk 15]
```diff
@@ -128,18 +128,18 @@ public class Product: Encodable {
...
     /// A library's product can either be statically or dynamically linked.
     /// If possible, don't declare the type of library explicitly to let 
-    /// the Swift Package Manager choose between static or dynamic linking based
+    /// the Swift package manager choose between static or dynamic linking based
     /// on the preference of the package's consumer.
     ///
     /// - Parameters:
     ///     - name: The name of the library product.
     ///     - type: The optional type of the library that is used to determine how to link to the library.
-    ///         Leave this parameter unspecified to let to let the Swift Package Manager choose between static or dynamic linking (recommended).
```
Comment (neonichu) [inline Sources/PackageDescription/Product.swift:142]: same as above

File: Sources/PackageDescription/Product.swift [hunk 16]
```diff
@@ -150,7 +150,7 @@ public class Product: Encodable {
         return Library(name: name, type: type, targets: targets)
     }
 
-    /// Create an executable package product that clients can run.
```
Comment (neonichu) [inline Sources/PackageDescription/Product.swift:153]: I think this was actually wrong before, clients can't run executables

File: Sources/PackageDescription/Resource.swift [hunk 17]
```diff
@@ -8,12 +8,27 @@
...
+///
+/// If a Swift package declares a Swift tools version of 5.3 or later, it can include resource files.
+/// Similar to source code, Xcode scopes resources to a target. As a result,
+/// you need to put them into the folder that corresponds to the target they belong to.
+/// For example, any resources for the `MyLibrary` target need to reside in `Sources/MyLibrary`.
+/// Use subdirectories to organize your resource files in a way that makes it easy to identify and manage them.
+/// For example, put all resource files into a directory named `Resources`,
+/// resulting in all of your resource files residing at `Sources/MyLibrary/Resources`.
+///
+/// Per default, Xcode handles common resources types for Apple platforms automatically.
```
Comment (neonichu) [inline Sources/PackageDescription/Resource.swift:21]: We should rephrase this as "when using the Xcode build system" because that's also possible from SwiftPM (using `--build-system xcode`.

File: Sources/PackageDescription/Resource.swift [hunk 18]
```diff
@@ -8,12 +8,27 @@
...
+/// You don’t need to declare XIB files, storyboards, CoreData file types, and asset catalogs
```
Comment (neonichu) [inline Sources/PackageDescription/Resource.swift:22]: This reads like a closed list, but it isn't actually fully comprehensive and could also go out of sync easily in the future. I think it would be better to list these as examples, but make clear that there could be more.

File: Sources/PackageDescription/Resource.swift [hunk 19]
```diff
@@ -35,25 +50,38 @@ public struct Resource: Encodable {
         self.localization = localization
     }
 
-    /// Apply the platform-specific rule to the given path.
```
Comment (neonichu) [inline Sources/PackageDescription/Resource.swift:38]: I think using "a" sounds to me as if I could pick a rule from a list of rules. Maybe "Applies platform specific rules" would be better?

File: Sources/PackageDescription/SupportedPlatforms.swift [hunk 20]
```diff
@@ -34,33 +34,33 @@ public struct Platform: Encodable {
...
 
-    /// The WASI platform
+    /// The WebAssembly System Interface platform.
     @available(_PackageDescription, introduced: 5.3)
     public static let wasi: Platform = Platform(name: "wasi")
 }
 
 /// A platform that the Swift package supports.
 ///
-/// By default, the Swift Package Manager assigns a predefined minimum deployment
```
Comment (neonichu) [inline Sources/PackageDescription/SupportedPlatforms.swift:52]: same as above

File: Sources/PackageDescription/SupportedPlatforms.swift [hunk 21]
```diff
@@ -34,33 +34,33 @@ public struct Platform: Encodable {
...
+/// By default, the Swift package manager assigns a predefined minimum deployment
 /// version for each supported platforms unless you configure supported platforms using the `platforms`
 /// API. This predefined deployment version is the oldest deployment target
 /// version that the installed SDK supports for a given platform. One exception
 /// to this rule is macOS, for which the minimum deployment target version
 /// starts from 10.10. Packages can choose to configure the minimum deployment
 /// target version for a platform by using the APIs defined in this struct. The
-/// Swift Package Manager emits appropriate errors when an invalid value is
```
Comment (neonichu) [inline Sources/PackageDescription/SupportedPlatforms.swift:59]: same as above

File: Sources/PackageDescription/SupportedPlatforms.swift [hunk 22]
```diff
@@ -34,33 +34,33 @@ public struct Platform: Encodable {
...
+/// Swift package manager emits appropriate errors when an invalid value is
 /// provided for supported platforms, such as an empty array, multiple declarations
 /// for the same platform, or an invalid version specification.
 ///
-/// The Swift Package Manager will emit an error if a dependency is not
```
Comment (neonichu) [inline Sources/PackageDescription/SupportedPlatforms.swift:63]: same as above

File: Sources/PackageDescription/Target.swift [hunk 23]
```diff
@@ -48,19 +48,22 @@ public final class Target {
 
     /// The path of the target, relative to the package root.
     ///
-    /// If the path is `nil`, the Swift Package Manager looks for a targets source files at predefined search paths
```
Comment (neonichu) [inline Sources/PackageDescription/Target.swift:51]: same as above

File: Sources/PackageDescription/Target.swift [hunk 24]
```diff
@@ -48,19 +48,22 @@ public final class Target {
...
+    /// If the path is `nil`, the Swift package manager looks for a target's source files at predefined search paths
     /// and in a subdirectory with the target's name.
     ///
     /// The predefined search paths are the following directories under the package root:
     ///   - For regular targets: `Sources`, `Source`, `src`, and `srcs`
     ///   - For test targets: `Tests`, `Sources`, `Source`, `src`, `srcs`
     ///
-    /// For example, the Swift Package Manager will look for source files inside the `[PackageRoot]/Sources/[TargetName]` directory.
```
Comment (neonichu) [inline Sources/PackageDescription/Target.swift:58]: same as above

File: Sources/PackageDescription/Target.swift [hunk 25]
```diff
@@ -147,7 +150,7 @@ public final class Target {
     }
     private var _linkerSettings: [LinkerSetting]?
 
-    /// The binary target's checksum.
```
Comment (neonichu) [inline Sources/PackageDescription/Target.swift:150]: I think it would be better if we kept this more generic, it could easily go out of sync if we add support for more formats. I think "ZIP file that contains the referenced binary artifact" or even more generic "archive file that..." would be preferable.

File: Sources/PackageDescription/Target.swift [hunk 26]
```diff
@@ -230,25 +233,26 @@ public final class Target {
...
-    ///   - exclude: A list of paths to files or directories that should not be considered source files. This path is relative to the target's directory.
+    ///   - path: The custom path for the target. By default, the Swift package manager expects a target's sources to reside at predefined search paths,
+    ///       for example, `[PackageRoot]/Sources/[TargetName]`.
+    ///       Don't escape the package root; that is, values like `../Foo` or `/Foo` are invalid.
+    ///   - exclude: A list of paths to files or directories that the Swift package manager shouldn't consider to be source or resource files.
+    ///       This path is relative to the target's directory.
     ///       This parameter has precedence over the `sources` parameter.
-    ///   - sources: An explicit list of source files. In case of a directory, the Swift Package Manager searches for valid source files
-    ///       recursively.
-    ///   - publicHeadersPath: The path to the directory containing public headers of a C-family target.
```
Comment (neonichu) [inline Sources/PackageDescription/Target.swift:251]: Not sure if we are actually restricting `publicHeadersPath` to just library targets.

File: Sources/PackageDescription/Target.swift [hunk 27]
```diff
@@ -535,15 +544,16 @@ public final class Target {
...
 
-    /// Create a binary target referencing a remote artifact.
+    /// Creates a binary target that references a remote artifact.
     ///
     /// A binary target provides the url to a pre-built binary artifact for the target. Currently only supports
     /// artifacts for Apple platforms.
     ///
     /// - Parameters:
     ///   - name: The name of the target.
-    ///   - url: The URL to the binary artifact. Should point to a `zip` archive containing a `XCFramework`.
```
Comment (neonichu) [inline Sources/PackageDescription/Target.swift:545]: Again not sure if we should mention XCFrameworks specifically here.

File: Sources/PackageDescription/BuildSettings.swift (hunk 1 above)
Comment (ElektrojungeAtWork) [reply Sources/PackageDescription/BuildSettings.swift:136]: Fixed. Now using title case.

File: Sources/PackageDescription/PackageDependency.swift (hunk 7 above)
Comment (ElektrojungeAtWork) [reply Sources/PackageDescription/PackageDependency.swift:228]: Using contractions follows the Apple style guide.

> “As part of Apple’s informal voice, contractions are used and recommended throughout most documentation, interface text, and marketing copy. [...]”

File: Sources/PackageDescription/PackageDescription.swift (hunk 8 above)
Comment (ElektrojungeAtWork) [reply Sources/PackageDescription/PackageDescription.swift:26]: True!

File: Sources/PackageDescription/PackageDescription.swift (hunk 9 above)
Comment (ElektrojungeAtWork) [reply Sources/PackageDescription/PackageDescription.swift:59]: We're not using code voice for names of frameworks as per style guide.

File: Sources/PackageDescription/Target.swift (hunk 26 above)
Comment (ElektrojungeAtWork) [reply Sources/PackageDescription/Target.swift:251]: It would make sense. Should I add this info?

**Comment by ElektrojungeAtWork** [comment]
The recent updates address the feedback. I've removed references to XCFramework, and made sure we're using title case when referring to the SwiftPM.

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
Oh, just saw that this is targeting the release/5.3 branch, we need a corresponding PR to master first.
